exercises/guillaume_communie/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * Skeleton main routine
+ */
+
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+#include <cctype>
+#include <iomanip>
+#include <set>
+
+const std::string RESULTS_FILENAME("word_counts.data");
+
+/**
+ * Print the program usage.
+ */
+void usage(void) {
+  std::cout << "Usage: WordCounter <file_name>" << std::endl;
+  exit(1);
+}
+
+/**
+ * Check if a characters is an 'interword' character.
+ * @param[in] c The character to test
+ * @return      True if the character is an interword character
+ */
+bool isInterWord(int c) {
+    return (c == ' ') || (c == '.') || (c == ',') || (c == '?') || (c == '\'')
+        || (c == '"') || (c == '!') || (c == '(') || (c == ')') || (c == ':')
+        || (c == '\n') || (c== '-');
+}
+
+int main(int argc, char *argv[])
+{
+    // check arguments
+    if (argc != 2)
+        usage();
+
+    // try to open the file
+    std::ifstream f(argv[1]);
+    if (!f.good()) {
+        std::cout << "Error: please provide a valid file path." << std::endl;
+        usage();
+    }
+
+    // read the words
+    std::map<std::string, int> dictionnary;
+    int character;
+    std::string word;
+    do {
+        character = f.get();
+        if (isalpha(character)) {
+            word.push_back((char) tolower(character));
```

Comment:
It's a good idea to specify the namespace for the algorithms so that there is no ambiguity.

Reply:
Here I am using `isalpha()` and `tolower()` from `ctype.h`